tests/test_tools.py

```diff
@@ -942,6 +942,30 @@ def test_env_path_add(left, right, exp):
     assert exp == obs
 
 
+def test_env_path_add_replace_no_dupes_front_replace_existing():
+    # Test replaces without dupes when added to front when adding existing entry
+    path = EnvPath(["/home/wakka", "/home/wakka/bin"])
+    path.add("/home/wakka/bin", front=True, replace=True)
```

Comment:
Add a test with `pathlib.Path("/home/wakka/bin")` to make sure that works correctly (which is the original issue)